Bugfix/70733 add white backing in editor

dsas opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- Link this PR to its associated issue with an appropriate keyword: Closes, See, Follow up to, etc. -->
Closes https://github.com/WordPress/gutenberg/issues/70733

It's worth pointing out that this PR is a step forward to aligning GB with the front-end, but isn't likely to be completely exhaustive in reference to treatment of alpha transparency on site elements.

<!-- In a few words, what is the PR actually doing? -->

Adds white "backing layers" to the editor canvas so that semi-transparent site background colours blend accurately and match the frontend.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

The site editor canvas isn't always white - when inserting patterns, normal template editing or using the left hand navigation to change things. This causes transparent RGBA site backgrounds (e.g., #F5F6FE30) to blend against that background instead of the intended white, leading to a visible color mismatch between the editor and the frontend.

This change builds upon an unmerged PR #70737 that @R1shabh-Gupta started and @t-hamano reviewed.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

 * Added a new white background layer behind the editor canvas
 * Did something very similar for zoomed out mode

## Testing Instructions

 1. Open the Site Editor.
 2. In the left panel, open Styles, then go to Background.
 3. Set a background color with transparency (e.g. #F5F6FE30).
 4. Observe the canvas: the background should now blend with white, matching the frontend appearance.
 5. Go to the template list
 6. Observe the canvas: the background of each template in the grid should now blend with white, matching the frontend appearance.
 7. Edit a template
 8. Observe the canvas: the background colour should remain blended with white
 9. Open the inserter, and then choose Patterns
 10. Observe the canvas: the background colour should remain blended with white

## Screenshots or screencast <!-- if applicable -->

(Using a test background colour of `#503aa830`)

<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

| |Before|After|
|-|-|-|
| Left nav | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 26 54" src="https://github.com/user-attachments/assets/3164bc43-08ea-4679-8dfa-0b2775a254ae" /> | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 23 15" src="https://github.com/user-attachments/assets/2a89b4c9-ea35-4d11-9578-d475fc7b6200" />
| Template grid | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 43 00" src="https://github.com/user-attachments/assets/1dcdef0c-11ed-4c62-9e86-0fdc6b6c7338" /> | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 42 27" src="https://github.com/user-attachments/assets/9342c5b4-fe76-44fd-a06a-804015348dc7" /> | 
| Editor | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 27 10" src="https://github.com/user-attachments/assets/fe1e5db2-64af-4f38-a567-bac21b350076" /> | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 24 21" src="https://github.com/user-attachments/assets/03b43912-7804-4ffe-93d2-439a3331e480" />|
| Zoomed-out | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 27 33" src="https://github.com/user-attachments/assets/fa5c1f48-7e4d-4d64-83c1-d8715c665e29" /> | <img width="1079" height="785" alt="Screenshot 2025-09-09 at 23 25 11" src="https://github.com/user-attachments/assets/e7476948-852a-4bd1-a2e9-fe159eeff09b" />
| Part editing | <img width="1429" height="1038" alt="Screenshot 2025-10-13 at 17 12 40" src="https://github.com/user-attachments/assets/892f5857-9f66-47b4-aa8d-3eef5d888f63" />  |  <img width="1429" height="1038" alt="Screenshot 2025-10-13 at 17 13 32" src="https://github.com/user-attachments/assets/eba77633-68bf-48d3-842b-5933cd8891b8" /> |

R1shabh-Gupta committed:
fix: add white backing layer behind editor canvas

dsas committed:
Adapt the white backling layer fix to zoomed-out mode

dsas committed:
Ensure white backing layer for block previews

dsas commented on packages/block-editor/src/components/block-preview/content.scss:
```diff
@@ -3,6 +3,11 @@
 
+.block-editor-block-preview__content-iframe {
+	// Ensure a white backing layer for consistent semi-transparent color blending
+	background-color: $white;
```
There's a brief flash of the darker colour while this loads 🤔

ellatrix commented:
Why doesn’t setting the iframe to white work?

ellatrix commented:
Sorry I’m mobile atm. To what element are global styles backgrounds applied? Is it the body? If so, could we make the html element white?

t-hamano commented:
The global styles background is applied to the `body` element. If we make the HTML element white, it overrides the gray background color in zoom-out mode:

<img width="1293" height="784" alt="iframe" src="https://github.com/user-attachments/assets/2b85b73e-24a7-4b9b-adf6-cad3925e702a" />

ellatrix commented on packages/editor/src/components/visual-editor/style.scss:
```diff
@@ -7,6 +7,13 @@
+		position: absolute;
+		inset: 0;
+		background-color: $white;
+	}
```
Why is this added a second time? Maybe it needs a comment?

ellatrix replied:
Also isn't `.editor-visual-editor` outside the iframe? Why does it need to use a pseudo element here?

dsas replied:
The white background-color should be applied between the body element (which gets the theme background-color applied) and any Gutenberg chrome. This ensures the behaviour is as close as possible to the front-end site ([though technically speaking "white" here is UA dependent](https://drafts.csswg.org/css-backgrounds-3/#canvas-surface)).

The GB chrome is implemented in different ways in different modes. Focusing purely on background-color:
 1. The general editor sets `iframe.edit-site-visual-editor__editor-canvas` to be transparent and so the `.editor-visual-editor` background-color shines through. 
 2. The template part editor (e.g. header editing) sets some of its styling inside the iframe - it sets a background-color on `iframe.edit-site-visual-editor__editor-canvas html.block-editor-iframe__html`
 3. The style editing mode sets the iframe.edit-site-visual-editor__editor-canvas to transparent and so the `.edit-site-layout:not(.is-full-canvas) .editor-visual-editor` background-color shines through
 4. The zoomed-out mode (e.g. pattern inserter) sets some of its styling inside the iframe too - it sets a background-color on `html.edit-site-visual-editor__editor-canvas iframe.block-editor-iframe__html.is-zoomed-out` (as well as setting background-color:initial)

In short, it's repeated because the background in gutenberg is complicated and handled diffusely (though the new `--wp-editor-canvas-background` variable consolidates it somewhat). 

I considered the comment for the background-color above it to cover the explanation as to why it's repeated. 

----

I think you're right about this not needing to be use a pseudo element. The iframe itself could probably just be set below to use `$white`.

dsas committed:
Minimise the changes by setting the iframe background

dsas committed:
fixup! Ensure white backing layer for block previews

dsas committed:
Fix undefined css var --wp-editor-canvas-background

This variable was not defined inside the iframe, and hence the grey
background was not being set in zoomed-out mode.
